### ISSUE TITLE
WebSocket Receive causing stack overflow issue. 

### ISSUE THREAD

**Issue by gavinbell** [unosquare/embedio#242]
I've been getting stack overflow and out of memory exceptions when loading testing an embedio websocket. 

I tracked the problem down to a Receive() local function in WebSocket.cs and it recursively calling itself. 
```
                if (!result || _readyState == WebSocketState.Closed)
                 {
                        _exitReceiving?.Set();
                        return;
                  }

                    // Receive next asap because the Ping or Close needs a response to it.
                    Receive();
```

I found that changing the Receive() function to use a while loop, for me, prevented the issue from manifesting.

```
async Task Receive()
{
    try
    {
        var frame = await frameStream.ReadFrameAsync(this).ConfigureAwait(false);

        if (frame == null)
            return;

        var result = await ProcessReceivedFrame(frame).ConfigureAwait(false);

        if (!result || _readyState == WebSocketState.Closed)
        {
            _exitReceiving?.Set();

            return;
        }

        var msgTask = Task.Run(Message);
    }
    catch (Exception ex)
    {
        Fatal("An exception has occurred while receiving.", ex);
    }
}

Task.Run(() =>
            {
                while (_readyState == WebSocketState.Open)
                    Receive().Wait();
            });
```

**Comment by geoperez** [comment]
Cool, I'm going to check and integrate it in the next release. thanks.

**Comment by geoperez** [comment]
I published your suggestion in version 2.2.7! Let me know if I can close this issue.
